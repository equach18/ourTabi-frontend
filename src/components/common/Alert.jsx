import React from "react";

/**
 * Presentational component for displaying alerts.
 *
 * Used in: Login, Signup, Profile Forms
 *
 * @param {string} type - "error" (default) or "success"
 * @param {string[]} messages - Array of messages to display
 */

function Alert({ type = "error", messages = [] }) {
  if (!messages.length) return null;

  const alertStyles =
    type === "error"
      ? "bg-red-100 text-red-700 border border-red-400"
      : "bg-green-100 text-green-700 border border-green-400";

  return (
    <div className={`p-3 rounded-lg ${alertStyles} my-2`}>
      {messages.map((msg, idx) => (
        <p className="text-sm" key={idx}>
          {msg}
        </p>
      ))}
    </div>
  );
}

export default Alert;
