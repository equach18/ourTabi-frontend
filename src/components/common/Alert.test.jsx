import React from "react";
import { render } from "@testing-library/react";
import Alert from "./Alert";
describe("Alert component", () => {
  it("renders without crashing", function () {
    render(<Alert />);
  });

  it("matches snapshot for default error alert", function () {
    const messages = ["Something went wrong."];
    const { asFragment } = render(<Alert messages={messages} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot for multiple error messages", function () {
    const messages = ["Error 1", "Error 2", "Error 3"];
    const { asFragment } = render(<Alert messages={messages} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot for success alert", function () {
    const messages = ["Action completed successfully."];
    const { asFragment } = render(<Alert type="success" messages={messages} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
