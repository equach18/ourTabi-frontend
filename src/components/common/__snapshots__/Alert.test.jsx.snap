// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert component > matches snapshot for default error alert 1`] = `
<DocumentFragment>
  <div
    class="p-3 rounded-lg bg-red-100 text-red-700 border border-red-400 my-2"
  >
    <p
      class="text-sm"
    >
      Something went wrong.
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert component > matches snapshot for multiple error messages 1`] = `
<DocumentFragment>
  <div
    class="p-3 rounded-lg bg-red-100 text-red-700 border border-red-400 my-2"
  >
    <p
      class="text-sm"
    >
      Error 1
    </p>
    <p
      class="text-sm"
    >
      Error 2
    </p>
    <p
      class="text-sm"
    >
      Error 3
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert component > matches snapshot for success alert 1`] = `
<DocumentFragment>
  <div
    class="p-3 rounded-lg bg-green-100 text-green-700 border border-green-400 my-2"
  >
    <p
      class="text-sm"
    >
      Action completed successfully.
    </p>
  </div>
</DocumentFragment>
`;
